src/operators/publishReplay.ts

```diff
@@ -3,10 +3,25 @@ import { ReplaySubject } from '../ReplaySubject';
 import { IScheduler } from '../Scheduler';
 import { multicast } from './multicast';
 import { ConnectableObservable } from '../observable/ConnectableObservable';
-import { UnaryFunction } from '../interfaces';
+import { UnaryFunction, MonoTypeOperatorFunction, OperatorFunction } from '../interfaces';
 
-export function publishReplay<T>(bufferSize: number = Number.POSITIVE_INFINITY,
-                                 windowTime: number = Number.POSITIVE_INFINITY,
-                                 scheduler?: IScheduler): UnaryFunction<Observable<T>, ConnectableObservable<T>> {
-  return (source: Observable<T>) => multicast(new ReplaySubject<T>(bufferSize, windowTime, scheduler))(source) as ConnectableObservable<T>;
+/* tslint:disable:max-line-length */
+export function publishReplay<T>(bufferSize?: number, windowTime?: number, scheduler?: IScheduler): UnaryFunction<Observable<T>, ConnectableObservable<T>>;
```

Comment:
The `publish` operator defines `MonoTypeOperatorFunction` return type but I don't think that's correct because this results into `Observable<T>` while `publish` should return `ConnectableObservable<T>` I think.

https://github.com/ReactiveX/rxjs/blob/master/src/operators/publish.ts#L6-L8

Reply:
This is fine for now... we might need to address this more generally across the library.